tls/x509utils/certpool/list.go

```diff
@@ -0,0 +1,126 @@
+package certpool
+
+import "container/list"
+
+// List is a typed wrapper on top of [list.List]
+type List[T comparable] list.List
+
+// Sys returns the native [list.List]
+func (l *List[_]) Sys() *list.List {
+	if l == nil {
+		return nil
+	}
+	return (*list.List)(l)
+}
+
+// Zero returns the zero value of the type associated
+// to the list
+func (*List[T]) Zero() T {
+	var zero T
+	return zero
+}
+
+// Append adds a non-zero value at the end of the list
+func (l *List[T]) Append(v T) {
+	if v != l.Zero() {
+		l.Sys().PushBack(v)
+	}
+}
+
+// First returns the first value in the list
+func (l *List[T]) First() T {
+	if l != nil {
+		if p := l.Sys().Front(); p != nil {
+			if v, ok := p.Value.(T); ok {
+				return v
+			}
+		}
+	}
+
+	return l.Zero()
+}
+
+// Last returns the last value in the list
+func (l *List[T]) Last() T {
+	if l != nil {
+		if p := l.Sys().Back(); p != nil {
+			if v, ok := p.Value.(T); ok {
+				return v
+			}
+		}
+	}
+
+	return l.Zero()
+}
+
+// Values returns all values in the list
+func (l *List[T]) Values() []T {
+	var out []T
+	l.ForEach(func(v T) bool {
+		out = append(out, v)
+		return true
+	})
+	return out
+}
+
+// ForEachElement safely calls a function on each element of the list.
```

Comment:
This looks to me a bit misleading. My first though was that `safe` refers to thread safe or goroutine safe. Only later I observed that `safe` means that you can call it with `nil` parameters and it will not `panic`. I would like to see a less misleading wording

Reply:
> This looks to me a bit misleading. My first though was that `safe` refers to thread safe or goroutine safe. Only later I observed that `safe` means that you can call it with `nil` parameters and it will not `panic`. I would like to see a less misleading wording

nah, _safe_ refers to the ability to remove entries while iterating (getting `next` before calling the function)

Reply:
dropped